src/advanced_inv.cpp

```diff
@@ -361,13 +362,22 @@ void advanced_inventory::print_items( advanced_inventory_pane &pane, bool active
         mvwprintz( window, 6 + x, weight_startpos, print_color, "%5.*f", w_precision, it_weight );
 
         //print volume column
-        int it_vol = sitem.volume / units::legacy_volume_factor;
+        double it_vol = units::to_liter( sitem.volume );
+        size_t v_precision;
         print_color = ( it_vol > 0 ) ? thiscolor : thiscolordark;
-        if( it_vol > 9999 ) {
-            it_vol = 9999;
-            print_color = selected ? hilite( c_red ) : c_red;
+        if( it_vol >= 100 ) {
+            if( it_vol >= 10000 ) {
+                it_vol = 9999;
+                print_color = selected ? hilite( c_red ) : c_red;
+            }
+            v_precision = 0;
+        } else if( it_vol >= 10 ) {
```

Comment:
This causes problems with alignment of the decimal point when you have for example `13.94` and `100.6`. We should eventually support toggling of displayed precision, possibly with automatic default, but I'd avoid such logic here for now